kernel/src/path.rs

```diff
@@ -1,34 +1,94 @@
 //! Utilities to make working with directory and file paths easier
 
-use lazy_static::lazy_static;
-use regex::Regex;
 use url::Url;
 
 use crate::{DeltaResult, Version};
 
-/// The delimiter to separate object namespaces, creating a directory structure.
+/// The delimiter to separate object namespaces, creating a directory structure. Note this is in url
+/// terms, so we use `/`
 const DELIMITER: &str = "/";
 
-lazy_static! {
-    static ref CHECKPOINT_FILE_PATTERN: Regex =
-        Regex::new(r"\d+\.checkpoint(\.\d+\.\d+)?\.parquet").unwrap();
-    static ref DELTA_FILE_PATTERN: Regex = Regex::new(r"\d+\.json").unwrap();
+#[derive(Debug)]
+pub(crate) struct LogPath<'a> {
+    url: &'a Url,
+    path: &'a str,
+    version: Option<Version>,
+    is_commit: bool,
+    is_checkpoint: bool,
 }
 
-#[derive(Debug)]
-pub(crate) struct LogPath<'a>(pub(crate) &'a Url);
+fn get_filename(path: &str) -> Option<&str> {
+    if path.is_empty() || path.ends_with('/') {
+        None
+    } else {
+        path.rsplit(DELIMITER).next()
+    }
+}
+
+fn try_get_version(version_str_opt: Option<&str>, expected_digits: usize) -> Option<Version> {
```

Comment:
nit: somehow `try_*` makes me think `Result`. A think I believe to be seeing more frequently is using `*_opt`, so maybe `get_version_opt`?